File: tests/fixtures/parity/csharp/dynamic_ref.cs
using System;
using System.Collections.Generic;

class Program {
    static Dictionary<string, Action> handlers = new Dictionary<string, Action>();

    static void Register(string name, Action fn) {
        handlers[name] = fn;
    }

    static void Main(string[] args) {
        Register("test", () => Console.WriteLine("Handler"));
        handlers["test"]();
    }
}

File: tests/fixtures/parity/java/dynamic_ref.java
import java.util.HashMap;
import java.util.Map;

public class Main {
    interface Handler {
        void handle();
    }

    static Map<String, Handler> handlers = new HashMap<>();

    static void register(String name, Handler handler) {
        handlers.put(name, handler);
    }

    public static void main(String[] args) {
        register("test", () -> System.out.println("Handler"));
        handlers.get("test").handle();
    }
}
